src/Definition/Source/DefinitionArray.php

```diff
@@ -106,17 +106,8 @@ public function getDefinition(string $name)
             $key = preg_quote($key);
             $key = '#' . str_replace('\\' . self::WILDCARD, self::WILDCARD_PATTERN, $key) . '#';
             if (preg_match($key, $name, $matches) === 1) {
-                $definition = $this->normalizer->normalizeRootDefinition($definition, $name);
-
-                // For a class definition, we replace * in the class name with the matches
-                // *Interface -> *Impl => FooInterface -> FooImpl
-                if ($definition instanceof ObjectDefinition) {
-                    array_shift($matches);
-                    $definition->setClassName(
-                        $this->replaceWildcards($definition->getClassName(), $matches)
-                    );
-                    $definition = $this->normalizer->normalizeRootDefinition($definition, $name);
-                }
```

Comment:
I've reworked this code so that `$this->normalizer->normalizeRootDefinition($definition, $name)` isn't called twice. This is now the normalizer that takes care of replacing wildcards.